components/org.wso2.carbon.data.provider/src/main/java/org/wso2/carbon/data/provider/endpoint/DataProviderEndPoint.java

```diff
@@ -103,6 +124,42 @@ public static void onOpen(WebSocketConnection webSocketConnection) {
     @OnMessage
     public void onMessage(String message, WebSocketConnection webSocketConnection) {
         DataProviderConfigRoot dataProviderConfigRoot = new Gson().fromJson(message, DataProviderConfigRoot.class);
+        String authoringClassName;
+        try {
+            Map webSocketConfiguration
+                    = (Map)getDataProviderHelper().getConfigProvider().getConfigurationObject(WEB_SOCKET_CONFIG_HEADER);
+            if (webSocketConfiguration != null) {
+                JSONObject webSocketConfigJSON = new JSONObject(new Gson().toJson(webSocketConfiguration));
+                if (!webSocketConfigJSON.has(WEB_SOCKET_AUTHORIZING_CLASS_CONFIG_HEADER)) {
+                    throw new Exception("Web socket authorizing class cannot be found in the deployment.yaml file.");
```

Comment:
Isn't this the same exception that get caught in the lines below.

Reply:
The reasoning why we cannot close is described in the previous review comment reply.